randovania/games/dread/exporter/patch_data_factory.py

```diff
@@ -117,18 +120,52 @@ def _node_for(self, identifier: AreaIdentifier | NodeIdentifier) -> Node:
 
     def _key_error_for_node(self, node: Node, err: KeyError):
         return KeyError(f"{self.game.world_list.node_name(node, with_world=True)} has no extra {err}")
+    
+    def _key_error_for_start_node(self, node: Node):
+        return KeyError(f"{self.game.world_list.node_name(node, with_world=True)} has neither a " + 
+                        "start_point_actor_name nor the area has a collision_camera_name for a custom start point")
 
     def _start_point_ref_for(self, node: Node) -> dict:
         world = self.game.world_list.nodes_to_world(node)
+        area = self.game.world_list.area_by_area_location(node.identifier.area_identifier)
         level_name: str = os.path.splitext(os.path.split(world.extra["asset_id"])[1])[0]
 
         try:
-            return {
-                "scenario": level_name,
-                "actor": node.extra["start_point_actor_name"],
-            }
-        except KeyError as e:
-            raise self._key_error_for_node(node, e)
+            if "start_point_actor_name" in node.extra:
+                return {
+                    "scenario": level_name,
+                    "actor": node.extra["start_point_actor_name"],
+                }
+            else:
+                collision_camera_name = area.extra["asset_id"]
+                spawnpoint_key = f"{level_name}_{collision_camera_name}_{node.location.x}" + \
```

Comment:
@ArcanoxDragon Was there a specific reason we use such a complicated key and not just the NodeIdentifier? Afaik something containing the world, area and node name should already be unique.

Reply:
The actor we create can literally be a static string, `SP_Randovania`. We'll never need to create more than one starting point for one given seed.

Reply:
This comes from a plando where we had one-way teleporter/transporter/elevator to custom points.
I could remove it and make it more simple or leave it for the opportunity to implement such a feature in the very far future.

Reply:
I'd say keep it in; the option for arbitrary teleport points is very beneficial for plando reasons. I don't recall why I chose the particular key format but if it can be simplified, I'm all for it.

Reply:
Then they're named based on the source teleporter at that point. Though could just make it SP_Randovania{i} with a helper that creates these.

Reply:
Sorry, I really can't follow your last message. 
`_start_point_ref_for` is called for the target and not for the source. The purpose is to check and prevent creating multiple times the same start point for the same node. e.g. two teleporters have the same target or one teleporter and the start location referencing the same node etc.
https://github.com/randovania/randovania/blob/d400cc33aef9686f5bdd699c703d41d09679bfd2/randovania/games/dread/exporter/patch_data_factory.py#L425-L431

Currently they are named `SP_RDV_i` where `i` is replaced by three digits number. You want to change `spawnpoint_name_prefix` to be a format string and move the logic in `_start_point_ref_for` to a helper function? Otherwise I don't see a difference from your suggestion compared to the current state.
https://github.com/randovania/randovania/blob/d400cc33aef9686f5bdd699c703d41d09679bfd2/randovania/games/dread/exporter/patch_data_factory.py#L145

It somehow sounds like you misread something like that the spawn points are named by teleporter name or similiar.

Reply:
One: you should have a helper method that takes a node, ensures a spawn point was created for it and returns the name.
Two: Why not just the node itself as a key for this?

Reply:
One: Alright.
Two: See above. That was my first message here to Arcanox because I didn't remember if there was a reason for such a complicated key.
Tests: Will be added for the gui and for the case you mentioned.

Reply:
Yeah, if this key is used purely in python there's no reason it can't just use a python object as key. Preferable to use, even.